src/runtime/hooks/iteration_hook.py

```
@@ -0,0 +1,196 @@
+        return []
+
+
+def format_iteration_context(learnings: list, current_iteration: int) -> str:
+    """Format previous iteration learnings for injection."""
+    if not learnings:
+        return ""
+
+    output = [f"# Previous Iteration Learnings (for Iteration {current_iteration})"]
+    output.append("")
+
+    for l in learnings[:5]:  # Max 5 previous iterations
```

Ruff (0.14.10)

Ambiguous variable name: `l`

(E741)